src/clingexplaid/preprocessors/preprocessor_assumption.py

```diff
@@ -0,0 +1,210 @@
+"""
+Transformer Module: Assumption Transformer for converting facts to choices that can be assumed
+"""
+
+import warnings
+from dataclasses import dataclass
+from typing import Dict, Iterable, List, Optional, Set, Tuple, Union
+
+import clingo
+from clingo.ast import ProgramBuilder, parse_string
+
+from ..exceptions import UnprocessedException
+from ..utils.match import match
+
+
+@dataclass
+class FilterSignature:
+    """Filters the facts converted to assumptions by signature"""
+
+    name: str
+    arity: int
+
+    def __hash__(self) -> int:
+        return hash((self.name, self.arity))
+
+
+@dataclass
+class FilterPattern:
+    """Filters the facts converted to assumptions by pattern"""
+
+    pattern: str
+
+    def __hash__(self) -> int:
+        return hash(self.pattern)
+
+
+class AssumptionPreprocessor:
+    """
+    A transformer that transforms facts that match with one of the signatures provided (no signatures means all facts)
+    into choice rules and also provides the according assumptions for them.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    def __init__(
+        self,
+        filters: Optional[Iterable[Union[FilterPattern, FilterSignature]]] = None,
+        control: Optional[clingo.Control] = None,
+        fail_on_unprocessed: bool = True,
+    ):
+        self.control = control if control is not None else clingo.Control()
+        self.filters: Set[Union[FilterPattern, FilterSignature]] = set(filters) if filters is not None else set()
+        self._filters_convert_nothing = bool(filters is not None and len(list(filters)) == 0)
```

Comment:
I think we mentioned that filters=[] is keeping all facts as facts since none if them will match "ANY" in an empty list, whereas filters=None is no filters needed, everything is transformed to a choice


if filters=None, then line 52 makes filters=set() and then. you get _filters_convert_nothing
if filters=[] then line 52 makes filters=set() and also you get _filters_convert_nothing

I am missing something?